test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightDeltaWriterTests.cs

```diff
@@ -1431,17 +1431,29 @@ public void WriteContentIn41DeletedEntry(bool isResponse)
                 );
         }
 
-        [InlineData(/*isResponse*/true)]
-        [InlineData(/*isResponse*/false)]
-        [Theory]
-        public void WriteDeletedEntryWithoutKeyOrIdShouldFail(bool isResponse)
+        [Fact]
+        public void WriteDeletedEntryWithoutKeyOrIdShouldNotFailWhenWritingAResponse()
+        {
+            this.TestInit(this.GetModel());
+            Action writeAction = () =>
+            {
+                ODataJsonLightWriter writer = new ODataJsonLightWriter(GetV401OutputContext(true), this.GetCustomers(), this.GetCustomerType(), true, false, true);
+                writer.WriteStart(true ? feed : requestFeed);
+                writer.WriteStart(new ODataDeletedResource());
+            };
+
+            writeAction.DoesNotThrow();
+        }
+
+        [Fact]
+        public void WriteDeletedEntryWithoutKeyOrIdShouldFailWhenWritingARequest()
         {
             this.TestInit(this.GetModel());
 
             Action writeAction = () =>
             {
-                ODataJsonLightWriter writer = new ODataJsonLightWriter(GetV401OutputContext(isResponse), this.GetCustomers(), this.GetCustomerType(), true, false, true);
-                writer.WriteStart(isResponse ? feed : requestFeed);
+                ODataJsonLightWriter writer = new ODataJsonLightWriter(GetV401OutputContext(false), this.GetCustomers(), this.GetCustomerType(), true, false, true);
+                writer.WriteStart(false ? feed : requestFeed);

```

Comment:
@ElizabethOkerio wont this always be false?

Reply:
yes. I think what needed to be done here was:  `writer.WriteStart(requestFeed)` or writer.WriteStart(feed).